kie-api/src/main/java/org/kie/api/io/ResourceType.java

```diff
@@ -30,24 +30,28 @@
 
     private static final long serialVersionUID = 1613735834228581906L;
 
-    private String name;
+    private final String name;
 
-    private String description;
+    private final String description;
 
-    private String defaultExtension;
+    private final String defaultExtension;
 
-    private String[] otherExtensions;
+    private final String[] otherExtensions;
 
-    private String defaultPath;
+    private final String defaultPath;
+
+    private final boolean isNative;
```

Comment:
Can you please clarify what `native` means in this context?
Why DMN, FEEL and PMML are the only type not native? What about `JAVA`, `BPMN` and similar?

Minor note: it should be better to remove `is` prefix because otherwise the getter should be named `isIsNative` to be compliant with the naming conventions :)

Reply:
In this context `native` means something that it is build natively by drools-compiler without going through an assembler. Also note that `native` is a Java keyword and that's why I named the field `isNative`. I'm not entirely happy with the name, so any suggestion for a better one is very welcome.